### DISCUSSION TITLE
Getting transaction_id as an identifier instead of original_transaciton_id in Appstore

### DISCUSSION THREAD

**Discussion by bishwadeep** [imdhemy/laravel-in-app-purchases#77] [General]
Hey mate,
Thank you for all the works you have done. It is really saving everyones time a lot. 
Is there a way I can use transaction_id instead of original_transaction_id as an identifier?

I could not see any functions that I can use to get the transaciton_id in AppStoreSubscription.php. The function is in ReceiptInfo.php 
`/**
     * @return string
     */
    public function getTransactionId(): string
    {
        return $this->transactionId;
    }`

Is there any way I can call it on the Listeners?

**Comment by imdhemy** [comment]
Hi @bishwadeep 

You can use the `getProviderRepresentation()` method from the `AppStoreSubscription`, it should return the full representation returned from the App Store and using it you can access any key using the associated getter method.

---
**Fun fact**:
I started to answer your questions yesterday but the electricity cut out twice 😅

**Reply by bishwadeep** [reply]
Hey Mate,
sorry for the late reply, thank you for the reply.
